src/opcodes/mod.rs

```diff
@@ -2370,10 +2375,401 @@ mod test {
             accumulator: 0x00,
             ..cpu_initial
         };
-        cpu_expected.set_flag(StatusFlag::Zero, true);
+        cpu_expected.set_flag(StatusFlag::Zero, false);
+        cpu_expected.set_flag(StatusFlag::Negative, true);
+
+        adc(&mut cpu_initial, &mut stub_bus, AddressModeValue::AbsoluteAddress(0x00ff));
+
+        assert_eq!(cpu_initial, cpu_expected);
+    }
+
+    #[test]
+    #[cfg(feature = "binary_coded_decimal")]
+    fn test_adc_decimal_1() {
+        const input: (u8, u8, bool) = (0x00, 0x00, false);
+        const expected: (u8, bool, bool, bool, bool) = (0x00, false, false, true, false);
+
+        let mut cpu_initial = MOS6502 {
+            accumulator: input.0,
+            x_register: 0x00,
+            y_register: 0x00,
+            program_counter: 0x0000,
+            stack_pointer: 0xfd,
+            status_register: 0x00,
+            ..Default::default()
+        };
+        cpu_initial.set_flag(StatusFlag::Decimal, true);
+        cpu_initial.set_flag(StatusFlag::Carry, input.2);
+
+        let mut stub_bus = StubInterface6502 {
+            read: |address, read_count| input.1,
+            write: |address, data, write_count| {
+                panic! {"Write function was called"}
+            },
+            ..Default::default()
+        };
+
+        let mut cpu_expected = MOS6502 {
+            accumulator: expected.0,
+            ..cpu_initial.clone()
```

Comment:
Clippy is complaining that the clone is redundant and failing the CI. Same goes for the other tests.

```suggestion
            ..cpu_initial
```